clouddq/classes/dq_rule_dimensions.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""todo: add classes docstring."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import clouddq.classes
+
+
+@dataclass
+class DqRuleDimensions:
+    """ """
+
+    dimensions: list | None = None
+
+    @classmethod
+    def update_config(
+        cls: DqRuleDimensions, config_current: dict, config_new: dict
+    ) -> dict:
+        return clouddq.classes.update_config_lists(config_current, config_new)
```

Comment:
on second thoughts, maybe update_config should be a classmethod for DqConfigType instead? It's not really used anywhere else in any of the classes and not updating any of the attributes so it's confusing to me to have this as part of the configs classes directly.

Reply:
It's a design choice, and the right answer depends on the expected changes. A good principle is to try to group code that is changing: For example, if we frequently add a concepts like "rule dimensions", a good design would allow you to add a class for it and implement all the specifics in that class. Nothing else changes. If you frequently make changes to how configuration is stored and loaded, it makes sense to group all that functionality in `DqConfigsCache`, for example, so that even if the whole underlying storage layer changes, changes are limited to one class.

Reply:
For me it makes sense where it is (unsurprising, since I put it there...): You have a configuration for `X` that you want to update, so you call `X.update_config` (where `X` is a rule, rule binding, entity...). Updating a config is not really related to `DqConfigsType` I think but it would make sense in `DqConfigsCache` which handles configuration? And in there you would have s/th like `if config_type == DqRuleDimensions then A else B`

Reply:
Moved it to `DqConfigsCache`